engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializer.scala

```diff
@@ -34,8 +34,8 @@ class ConfluentKafkaAvroDeserializer[T](kafkaConfig: KafkaConfig, schemaData: Ru
     KryoGenericRecordSchemaIdSerializationSupport.schemaIdSerializationEnabled(kafkaConfig)
 
   override def deserialize(topic: String, data: Array[Byte]): T = {
-    val record = deserialize(topic, isKey, data, schemaData)
-    record.asInstanceOf[T]
+    val payloadObj = deserialize(topic, isKey, data, schemaData)
```

Comment:
It is not always kafka payload. It can be used also for `isKey = true`.

Reply:
Neither it is a "record", which is somewhat misleading. Changed name to neutral `deserializedData` (to match parameter `data`).